react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js

```diff
@@ -3,21 +3,31 @@ import { actions } from './actions';
 const initialState = {
   books: [],
   bookSelected: [],
-  originalData: []
+  originalData: [],
+  booksQuantity: {}
 };
 
 function reducer(state = initialState, action) {
-  switch (action.type) {
-    case actions.GET_BOOKS: // TODO to implement the logic
-      return { ...state };
-    case actions.ADD_TO_CART: // TODO to implement the logic
-      return { ...state };
-    case actions.ADD_ITEM: // TODO to implement the logic
-      return { ...state };
-    case actions.REMOVE_ITEM: // TODO to implement the logic
-      return { ...state };
-    case actions.SEARCH_ITEM: // TODO to implement the logic
-      return { ...state };
+  const { payload, type } = action;
+  switch (type) {
+    case actions.GET_BOOKS:
+      payload.forEach(book => {
+        if (!(book.id in state.booksQuantity)) state.booksQuantity = { ...state.booksQuantity, [book.id]: 1 };
```

Comment:
don't use one line if's as the style guide recommends

Reply:
also you are mutating the stage here.
Also i dont understand the `booksQuantity` variable name, because quantity always refeers to a number, but here you are using an object